pypeit/core/wavecal/autoid.py

```diff
@@ -688,6 +695,129 @@ def reidentify(spec, spec_arxiv_in, wave_soln_arxiv_in, line_list, nreid_min, de
+            they cannot be accurately centroided
+        sigdetect (float, optional): Defaults to 5.0.
+            Threshold for detecting arcliens
+        fwhm (float, optional): Defaults to 4.0.
+            Full width at half maximum for the arc lines
+        debug_peaks (bool, optional): Defaults to False.
+        use_unknowns (bool, optional): Defaults to False.
+            If True, use the unknowns in the solution (not recommended)
+
+    Returns:
+        tuple: tcent (np.ndarray; centroid of lines), spec_cont_sub (np.ndarray; subtracted continuum), 
+            patt_dict_slit (dict; dictionary on the lines), tot_line_list (astropy.table.Table; line list)
+    """
+    # Load line list
+    tot_line_list, _, _ = waveio.load_line_lists(lamps, include_unknown=use_unknowns)
+
+
+    # Generate the wavelengths from the line list and sort
+    wvdata = np.array(tot_line_list['wave'].data)  # Removes mask if any
+    wvdata.sort()
+
+    # Search for lines in the input arc
+    tcent, ecent, cut_tcent, icut, spec_cont_sub = wvutils.arc_lines_from_spec(
+        spec, sigdetect=sigdetect,
+        nonlinear_counts=nonlinear_counts, 
+        fwhm=fwhm, debug = debug_peaks)
+    # If there are no lines in the input arc, return
+    if tcent.size == 0:
+        return None
+
+    # Search for lines in the arxiv arc
+    tcent_arxiv, ecent_arxiv, cut_tcent_arxiv, icut_arxiv, spec_cont_sub_now = wvutils.arc_lines_from_spec(
+            spec_arxiv, sigdetect=sigdetect,
+            nonlinear_counts=nonlinear_counts, fwhm = fwhm, debug = debug_peaks)
+    # If there are no lines in the arxiv arc, return
+    if tcent_arxiv.size == 0:
+        return None
+
+    # Interpolate the input wavelengths
+    fwv_guess = scipy.interpolate.interp1d(np.arange(len(wv_guess)), wv_guess, 
+                                   kind='cubic', bounds_error=False, 
+                                   fill_value='extrapolate')
+    # Interpolate the arxiv both ways
+    fpix_arxiv = scipy.interpolate.interp1d(wave_arxiv, np.arange(len(wave_arxiv)), 
+                                   kind='cubic', bounds_error=False, 
+                                   fill_value='extrapolate')
+    fwv_arxiv = scipy.interpolate.interp1d(np.arange(len(wave_arxiv)), wave_arxiv, 
+                                   kind='cubic', bounds_error=False, 
+                                   fill_value='extrapolate')
+    # Find the wavelengths of the input arc lines and then the pixels
+    wv_cent = fwv_guess(tcent)
+    pix_arxiv = fpix_arxiv(wv_cent)
+
+    # Other bits
+    wvc_arxiv = wave_arxiv[wave_arxiv.size//2]
+    igood = wave_arxiv > 1.0
+    disp_arxiv = np.median(wave_arxiv[igood] - np.roll(wave_arxiv[igood], 1))
+
+    line_indx = np.array([], dtype=int)
+    det_indx = np.array([], dtype=int)
+    line_cc = np.array([], dtype=float)
+    #line_iarxiv = np.array([], dtype=int)
+
+    # Match with tolerance
```

Comment:
A lot of this code below looks like it is adapted from lines 632-646 of reidentify (on your branch). Is there a way to consolidate the two control flows into one method so that we don't have to maintain duplicated pieces of code?

Reply:
yeah, I looked at this and was initially on that path but the inputs are
control flow are sufficiently different that this became the cleaner soultion
despite the duplicated code.